feat: Add wallet balance telegraf probe

pyropy opened the pull request:
Adds a telegraf probe for checking wallet balance.

pyropy committed:
feat: Add wallet balance telegraf probe

juliangruber commented on probes/wallet-balance.sh:
```diff
@@ -0,0 +1,70 @@
+  RPC_URL="https://api.node.glif.io/rpc/v1"
+fi
+
+tmp="$(mktemp)"
```
instead of making a temp dir, this can be one pipeline (curl | jq | printf)

pyropy replied:
I've removed the support for filecoin native address format in https://github.com/filbeam/telegraf/pull/7/commits/fa0c0281ff59729ea5a8faa1fe9abb11029a2bfd and in return there's no need for temp dir anymore making the script much cleaner 👍🏻

juliangruber reviewed (changes requested):
While this does look like a full featured solution, it adds features we don't need yet, which can later on break and cause bugs.

pyropy committed:
chore: simplify script

juliangruber commented on probes/wallet-balance.sh:
```diff
@@ -0,0 +1,32 @@
+BALANCE_HEX=$(curl -sS -X POST "$RPC_URL" \
+  -H 'Content-Type: application/json' \
+  --data "{\"jsonrpc\":\"2.0\",\"method\":\"eth_getBalance\",\"params\":[\"$WALLET_ADDRESS\",\"latest\"],\"id\":1}" \
+  | jq -r '.result // "0x0"')
```
this is missing a `GLIF_TOKEN`, missed that in the previous review

pyropy replied:
Seemed to work fine without it but I've added one in https://github.com/filbeam/telegraf/pull/7/commits/74ac2903659f419ef75508cfa2fda149d13edcdc

juliangruber replied:
It will work fine for a low request volume only. I'm not sure though what the limit is atm

pyropy committed:
chore: add glif token

juliangruber commented on telegraf.conf:
```diff
@@ -36,3 +36,12 @@
+[[inputs.exec]]
+  commands = [
+    "/opt/probes/wallet-balance.sh calibration ${CALIBRATION_FILBEAM_CONTROLLER_WALLET}",
+  ]
```
ins't this missing the GLIF_TOKEN env var? Or is that passed through automatically?

bajtos commented on telegraf.conf:
```diff
@@ -36,3 +36,12 @@
+[[inputs.exec]]
+  commands = [
+    "/opt/probes/wallet-balance.sh calibration ${CALIBRATION_FILBEAM_CONTROLLER_WALLET}",
+  ]
```
We will need to monitor the wallet balance of the mainnet controller wallet, too. Since FWSS v1.0.0 was already deployed to the mainnet, I propose to include the mainnet setup from the start.

```suggestion
  commands = [
    "/opt/probes/wallet-balance.sh calibration ${CALIBRATION_FILBEAM_CONTROLLER_WALLET}",
    "/opt/probes/wallet-balance.sh mainnet ${MAINNET_FILBEAM_CONTROLLER_WALLET}",
  ]
```

pyropy replied:
I had no idea that it was deployed to mainnet 👀

bajtos commented on probes/wallet-balance.sh:
In Spark, we used `inputs.http` to make the RPC calls. I am not opposed to using a shell script, but I am wondering why we are introducing a new approach. 🤔 

https://github.com/CheckerNetwork/telegraf/blob/main/telegraf.conf#L33-L45

Since the new shell script is already implemented and reviewed, I guess it's best to keep it. 💪🏻

pyropy committed:
Add controller mainnet wallet probe

Co-authored-by: Miroslav Bajtoš <oss@bajtos.net>

pyropy commented on telegraf.conf:
```diff
@@ -36,3 +36,12 @@
+    "/opt/probes/wallet-balance.sh calibration ${CALIBRATION_FILBEAM_CONTROLLER_WALLET}",
+  ]
+  data_format = "influx"
+  interval    = "5m"
```
SGTM

pyropy committed:
Reduce wallet probe interval

Reduce wallet probe interval from 5m to 1m

Co-authored-by: Miroslav Bajtoš <oss@bajtos.net>

juliangruber commented on probes/wallet-balance.sh:
```diff
@@ -0,0 +1,37 @@
+
+BALANCE=$(printf "%d" "$BALANCE_HEX")
+
+MEASUREMENT="wallet_balance"
```
good idea, should have a key so we can add multiple

pyropy committed:
feat: Add wallet name parameter to wallet balance measurement

- Update wallet-balance.sh to accept a wallet name as the second parameter
- Change measurement name from generic "wallet_balance" to "{wallet_name}_balance"
- Update telegraf.conf to pass "filbeam_controller" as the wallet name

Addresses PR comment: https://github.com/filbeam/telegraf/pull/7/files#r2486717510

pyropy committed:
chore: add wallet env variables

bajtos reviewed (approved):
:shipit:

pyropy committed:
refactor: use native http input instead of bash script
